L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc

```diff
@@ -281,6 +286,12 @@ L1FPGATrackProducer::L1FPGATrackProducer(edm::ParameterSet const& iConfig)
                                    << "\n table_TRE    :  " << tableTREFile.fullPath();
     }
   }
+
+  trackQuality_ = iConfig.getParameter<bool>("TrackQuality");
+  if (trackQuality_) {
+    trackQualityParams = iConfig.getParameter<edm::ParameterSet>("TrackQualityPSet");
+    trackQualityModel.reset(new TrackQuality(trackQualityParams));
```

Comment:
where does this memory gets deallocated ? Please use smart pointers instead.

For example:
`trackQualityModel.reset(std::make_unique<TrackQuality>(trackQualityParams));`

Reply:
From what I understand about smart pointers, trackQualityModel is defined as a smart pointer (L176) and when it is reset at L293 it deletes the old unique_ptr and takes ownership of the new one. When trackQualityModel is destroyed, the pointer managed by it is also deleted. Therefore, all of the memory should be deallocated when the trackQualityModel unique_ptr is destroyed. It's possible that I'm wrong as I am just learning about smart pointers myself.

The example you gave does not work for this reset function. It must be reset with a general "new" pointer as seen here. I also found an exchange that asks why it cannot be reset with another unique_ptr here.

Perhaps instead of reseting the unique_ptr, we can reinitialize it? I can replace L293 with
`trackQualityModel = std::make_unique<TrackQuality>(trackQualityParams);`
which avoids using "new". This method is perhaps preferred to reset(). Should I make this change instead?

Reply:
I made the change that I suggested above to make re-assign the variable to a smart pointer